packages/design-system/src/components/datalist/list/list.tsx

```diff
@@ -262,44 +267,34 @@ const OptionList = forwardRef(function OptionList(
                   <OptionRenderer
                     key={option.id || ''}
                     role={'option'}
-                    tabIndex="-1"
+                    tabIndex={-1}
                     aria-selected={value === option.id}
                     aria-posinset={optionInset + 1}
                     aria-setsize={filteredOptions.length}
                     data-option={option.id}
                     onClick={() => onSelect(option)}
-                    ref={(el) => (optionsRef.current[optionInset] = el)}
+                    ref={(el: HTMLLIElement | null) =>
+                      (optionsRef.current[optionInset] = el)
+                    }
                     option={option}
                     value={value}
                   />
                 );
               })}
-            </Group>
+            </GroupList>
           )
         );
       })}
     </List>
   );
-});
+}
 
-OptionList.propTypes = {
-  keyword: PropTypes.string,
-  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-  onSelect: PropTypes.func,
-  onClose: PropTypes.func,
-  onExpandedChange: PropTypes.func,
-  focusTrigger: PropTypes.number,
-  options: PropTypes.array,
-  primaryOptions: PropTypes.array.isRequired,
-  primaryLabel: PropTypes.string,
-  priorityOptionGroups: PropTypes.array,
-  searchResultsLabel: PropTypes.string,
-  renderer: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
-  onObserve: PropTypes.func,
-  listId: PropTypes.string.isRequired,
-  focusSearch: PropTypes.func,
-  listStyleOverrides: PropTypes.array,
-  noMatchesFoundLabel: PropTypes.string,
-};
+const OptionList = memo(forwardRef(OptionListWithRef)) as <
```

Comment:
🤯

Reply:
Yeah, I know. It sucks. I should probably repeat the comment about why this is necessary when having generically typed components.